activation/atxdb.go

```diff
@@ -366,111 +357,88 @@ func (db *ActivationDb) ActiveSetSize(epochId types.EpochId) (uint32, error) {
+	}
+	var topAtx atxIdAndLayer
+	err = types.BytesToInterface(topAtxBytes, &topAtx)
+	if err != nil {
+		return atxIdAndLayer{}, fmt.Errorf("failed to unmarshal top ATX: %v", err)
+	}
+	return topAtx, nil
 }
 
 func getNodeIdKey(id types.NodeId) []byte {
 	return []byte(id.Key)
 }
 
-// addAtxToNodeIdSorted inserts activation atx id by node it is not thread safe and needs to be called under a global lock
-func (db *ActivationDb) addAtxToNodeIdSorted(nodeId types.NodeId, atx *types.ActivationTx) error {
+// addAtxToNodeId inserts activation atx id by node
+func (db *ActivationDb) addAtxToNodeId(nodeId types.NodeId, atx *types.ActivationTx) error {
 	key := getNodeIdKey(nodeId)
-	ids, err := db.atxs.Get(key)
-	var atxs []types.AtxId
-	if err != nil {
-		//layer doesnt exist, need to insert new layer
-		ids = []byte{}
-		atxs = make([]types.AtxId, 0, 0)
-	} else {
-		atxs, err = decodeAtxIds(ids)
-		if err != nil {
-			return errors.New("could not get all atxs from database ")
-		}
-	}
-
-	// append needs to be sorted
-	atxs = append(atxs, atx.Id())
-	l := len(atxs)
-	if l > 1 {
-		lastAtx, err := db.getAtxUnlocked(atxs[l-2])
-		if err != nil {
-			return errors.New("could not get atx from database ")
-		}
-		if atx.Sequence < lastAtx.Sequence {
-			sort.Slice(atxs, func(i, j int) bool {
-				atx1, err := db.getAtxUnlocked(atxs[i])
-				if err != nil {
-					panic("inconsistent db")
-				}
-				atx2, err := db.getAtxUnlocked(atxs[j])
-				if err != nil {
-					panic("inconsistent db")
-				}
-
-				return atx1.Sequence < atx2.Sequence
-			})
-		}
-	}
-
-	w, err := encodeAtxIds(atxs)
+	err := db.atxs.Put(key, atx.Id().Bytes())
 	if err != nil {
-		return errors.New("could not encode layer atx ids")
+		return fmt.Errorf("failed to store ATX ID for node: %v", err)
 	}
-	return db.atxs.Put(key, w)
+	return nil
 }
 
-// GetNodeAtxIds returns the atx ids that were received for node nodeId
-func (db *ActivationDb) GetNodeAtxIds(nodeId types.NodeId) ([]types.AtxId, error) {
+// GetNodeLastAtxId returns the last atx id that was received for node nodeId
+func (db *ActivationDb) GetNodeLastAtxId(nodeId types.NodeId) (types.AtxId, error) {
 	key := getNodeIdKey(nodeId)
-	db.log.Debug("fetching atxIDs for node %v", nodeId.Key[:5])
+	db.log.Debug("fetching atxIDs for node %v", nodeId.ShortString())
 
-	db.RLock()
-	ids, err := db.atxs.Get(key)
-	db.RUnlock()
+	id, err := db.atxs.Get(key)
 	if err != nil {
-		return nil, err
-	}
-	atxs, err := decodeAtxIds(ids)
-	if err != nil {
-		return nil, err
+		return *types.EmptyAtxId, err
 	}
-	return atxs, nil
+	return types.AtxId{Hash: common.BytesToHash(id)}, nil
 }
 
-// GetEpochAtxIds returns all atx id's created in this epoch as seen by this node
-func (db *ActivationDb) GetEpochAtxIds(epochId types.EpochId) ([]types.AtxId, error) {
-	db.RLock()
-	ids, err := db.atxs.Get(epochId.ToBytes())
-	db.RUnlock()
+// GetPosAtxId returns the best (highest layer id), currently known to this node, pos atx id
```

Comment:
let's add here a todo to consider another way to select the positioning atx once we introduce ticks

Reply:
I kept this in mind when developing this part and the change should be minimal. I don't think a TODO to the code has much value because I won't remember to look for it.

What could be useful is to add a TODO to the ticks issue... or maybe that's what you meant?

I've created https://github.com/spacemeshos/go-spacemesh/issues/1257